react.md

```diff
@@ -0,0 +1,242 @@
+  render: ->
+    <div>Here is some content</div>
+    <div>
+      <p>Here is some more</p>
+    </div>
+
+  # GOOD EXAMPLE:
+  render: ->
+    <section>
+      <header>
+        <nav>Here is the navigation</nav>
+      </header>
+      <div>
+        <p>Here is the first text on the page!</p>
+      </div>
+    </section>
+  ```
+
+2. If you have more than 3 props on a React component, move each prop declaration to a new line
+
+  ```CoffeeScript
+  # Good
+  render: ->
+    <span>
+      <ComponentName a={@state.a} b={@state.b} />
+    </span>
+
+  # Bad
+  render: ->
+    <span>
+      <ComponentName a={@state.a} b={@state.b} c={@state.c} d={@state.d} />
+    </span>
+
+  # Good; note how the termination of the null component is on its own line
+  render: ->
+    <span>
+      <ComponentName
+        a={@state.a}
+        b={@state.b}
+        c={@state.c}
+        d={@state.d}
+      />
+    </span>
+
+  # Good; note how the termination of the component with `children` is on the same line
+  # as the last prop
+  render: ->
+    <span>
+      <ComponentName
+        something={@props.something}
+        otherStuff={@props.otherStuff}
+        thing={@state.thing}>
+        <span>Inner content!</span>
+      </ComponentName>
+    </span>
+  ```
+
+3. Keep one space before the termination of a void component (one that does not have `children`)
+
+  ```CoffeeScript
+  <ComponentName x={@state.x} /> # Good
+  <ComponentName x={@state.x}/> # Bad
+  ```
+
+4. Wrap all inline object literals in curly braces
+
+  ```CoffeeScript
+  <ComponentName x={y: 5} /> # Bad
+  <ComponentName x={{y: 5}} /> # Good
+  ```
+
+5. If you are implicitly passing down `props` or `state`, use the Spread operator:
+
+  ```CoffeeScript
+  # @state = {x: 5, y: 6, z: 7}
+  <ComponentName {...@state} /> # Then you can call `@prop.x` inside of `ComponentName`
+  ```
+
+6. Use multi-line if-statements wherever possible
+
+  ```CoffeeScript
+  render: ->
+    text =
+      if @state.isVisible
+        <span>Or this is alright, too</span>
+    <span>
+      {text} 
+      {if @state.isVisible then <h1>Show this</h1>} # Good, since it's short
+      {if @state.isVisible then <p><strong>Bold first word</strong> and regular 
+       elsewhere</p>} # Bad, because it's not short
+      {if @state.isVisible then {@state.text}} # Good, since it's short (and a variable)
+      {if @state.isVisible
+        <p><strong>Bold first word</strong> and regular stuff elsewhere</p>
+      } # Good, since it's long
+    </span>
+  ```
+
+7. If you have multiple mixins, define each one on a new line:
+
+  ```CoffeeScript
+  SomeComponent = React.createClass
+    mixins: [
+      MixinOne
+      MixinTwo
+    ]
+  ```
+
+8. Remember to `key` like elements that appear in arrays
+  1. Any adjacent React Components are represented by arrays of elements.
+  2. This is to provide React with the information to hueristically optimize UI reconciliation (figuring out what to re-render and what not to)
```

Comment:
Not sure. Technically, you could have a scenario like this:

```CoffeeScript
makeItem: (item, index) ->
  <li key={index}>{item}</li>
render: ->
  <ul>{@props.products.map(@makeItem)}</ul>
```

There's no way for a linter to know (at build time) the context of the `makeItem` method--whether it's called inside of `map` or called on it's own. 

My gut is that we're going to have to rely on console warnings & forcing habits for the time being--React can tell when this is going to be a problem when it runs in development mode.